scripts/west_commands/runners/openocd.py

```diff
@@ -369,16 +368,51 @@ def do_attach_debug(self, command, **kwargs):
         gdb_cmd = (self.gdb_cmd + self.tui_arg +
                    ['-ex', f'target extended-remote :{self.gdb_client_port}',
                     self.elf_name])
+
         if command == 'debug':
             gdb_cmd.extend(self.load_arg)
         if self.gdb_init is not None:
             for i in self.gdb_init:
                 gdb_cmd.append("-ex")
                 gdb_cmd.append(i)
+        if command == 'rtt':
+            rtt_address = self.get_rtt_address()
+            if rtt_address is None:
+                raise ValueError("RTT Control block not be found")
+
+            # gdb is not intended to be used interactively with rtt. It is only used to load the
+            # binary, so we cannot prompt the user to press return to see more text. Disable
+            # pagination for rtt.
+            gdb_cmd.extend(['-ex', 'set pagination off'])
+
+            # start the rtt server via gdb monitor commands (which are passed to openocd)
+            gdb_cmd.extend(
+                ['-ex', f'monitor rtt setup 0x{rtt_address:x} 0x10 "SEGGER RTT"'])
+            gdb_cmd.extend(['-ex', 'monitor reset run'])
+            gdb_cmd.extend(['-ex', 'monitor rtt start'])
+            gdb_cmd.extend(
+                ['-ex', f'monitor rtt server start {self.rtt_port} 0'])
+
+            # detach from the target and quit the gdb session after running the above commands
+            # so that the rtt client can connect to the server
+            gdb_cmd.extend(['-ex', 'detach', '-ex', 'quit'])
 
         self.require(gdb_cmd[0])
         self.print_gdbserver_message()
-        self.run_server_and_client(server_cmd, gdb_cmd)
+
+        if command in ('attach', 'debug'):
+            self.run_server_and_client(server_cmd, gdb_cmd)
+        elif command == 'rtt':
+            self.print_rttserver_message()
+            server_proc = self.popen_ignore_int(server_cmd)
```

Comment:
It would probably make sense to include this call in the `try` / `except` block below.

Reply:
Fixed!